src/script/script.cpp

```diff
@@ -347,7 +347,7 @@ bool GetScriptOp(CScriptBase::const_iterator& pc, CScriptBase::const_iterator en
 
 bool IsOpSuccess(const opcodetype& opcode)
 {
-    return opcode == 80 || opcode == 98 || (opcode >= 126 && opcode <= 129) ||
+    return opcode == 80 || opcode == 98 || (opcode >= 127 && opcode <= 129) ||
```

Comment:
This shouldn't be changed since the `opcode == OP_CAT` checks are conditional on `IsOpSuccess()` passing.

Reply:
Isn't it unnatural for CAT to still be "OpSuccess" after enabling it? I would think the check on script flags for OP_CAT should rather be moved out of the `IsOpSuccess` clause.

Reply:
I guess it depends: if you view `IsOpSuccess` as taking the following BIP342's definition of what `OP_SUCCESS` means, then it shouldn't change, but if you define it as the sum of BIP342 plus BIN-2024-001 (plus "LNHANCE", plus [64bitmath](https://github.com/bitcoin/bitcoin/pull/29265), plus...) then you'd want to change it. I think it's probably simplest just to leave it fixed as evaluating BIP342's definition, and have the conditional logic ("is OP_CAT active? then it's not OP_SUCCESSish") in the caller.